### PR TITLE
refactor(fs): prefer our fs and fs to fs-extra (pt2: test files)

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
Follow up from https://github.com/aws/aws-toolkit-vscode/pull/5689, focusing on the test files. 

## Solution
Mostly direct translation, one implementation change from `writeJson` to `Json.stringify` then `writeFile` in `packages/core/src/test/awsService/cdk/detectCdkProjects.test.ts` line 120

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]

**Comment by justinmk3** [comment]
this failure on the "minimum" unit test job might be relevant to this PR?

```
  3 failing
  1) CodeDownloader
       "after each" hook for "should return an error if the response body is not Buffer":
     EntryIsADirectory (FileSystemError): Error: EISDIR: illegal operation on a directory, unlink '/tmp/aws-toolkit-vscode/vsctk4b58ed57'
      at Function._handleError (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121929)
      at Object.delete (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121007)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (/codebuild/output/src373008039/src/github.com/aws/aws-toolkit-vscode/packages/core/src/test/eventSchemas/commands/downloadSchemaItemCode.test.ts:52:9)

  2) CodeGenerator
       "after each" hook for "should return the current status of code generation":
     EntryIsADirectory (FileSystemError): Error: EISDIR: illegal operation on a directory, unlink '/tmp/aws-toolkit-vscode/vsctkddbc90b6'
      at Function._handleError (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121929)
      at Object.delete (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121007)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (/codebuild/output/src373008039/src/github.com/aws/aws-toolkit-vscode/packages/core/src/test/eventSchemas/commands/downloadSchemaItemCode.test.ts:132:9)

  3) CodeGeneratorStatusPoller
       "after each" hook for "should return the current status of code generation":
     EntryIsADirectory (FileSystemError): Error: EISDIR: illegal operation on a directory, unlink '/tmp/aws-toolkit-vscode/vsctkc02d59ff'
      at Function._handleError (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121929)
      at Object.delete (/tmp/.vscode-test/vscode-linux-x64-1.68.0/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:83:121007)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Context.<anonymous> (/codebuild/output/src373008039/src/github.com/aws/aws-toolkit-vscode/packages/core/src/test/eventSchemas/commands/downloadSchemaItemCode.test.ts:205:9)
```

**Comment by Hweinstock** [comment]
Yep, looks fixed now. The last one is the one fixed here: https://github.com/aws/aws-toolkit-vscode/pull/5725

File: packages/core/src/test/shared/credentials/userCredentialsUtils.test.ts [hunk 1]
```diff
@@ -20,6 +19,16 @@ import { UserCredentialsUtils } from '../../../shared/credentials/userCredential
 import { EnvironmentVariables } from '../../../shared/environmentVariables'
 import { makeTemporaryToolkitFolder } from '../../../shared/filesystemUtilities'
 import { getConfigFilename, getCredentialsFilename } from '../../../auth/credentials/sharedCredentialsFile'
+import { fs } from '../../../shared'
+
+/** Async version of "doesNotThrow" */
+async function assertDoesNotThrow(fn: () => Promise<void>): Promise<void> {
```
Comment (justinmk3) [inline packages/core/src/test/shared/credentials/userCredentialsUtils.test.ts:25]: does `assert.doesNotReject` work?

**Pull request merged**